nemo/collections/tts/models/magpietts.py

```diff
@@ -3527,31 +3529,80 @@ def do_tts(
-        tokens = tokens + [self.eos_id]  # Add EOS token (BOS not used per dataset convention)
-        text_tensor = torch.tensor([tokens], device=self.device, dtype=torch.long)
-        text_lens = torch.tensor([len(tokens)], device=self.device, dtype=torch.long)
-
-        # Create batch dictionary
-        batch = {
-            'text': text_tensor,
-            'text_lens': text_lens,
-            'speaker_indices': speaker_index,
-        }
+        # Detect if longform inference is needed based on word count
+        word_count = len(normalized_text.split())
+        longform_word_threshold = 40
 
-        # Run inference
         with torch.no_grad():
-            output = self.infer_batch(
-                batch,
-                max_decoder_steps=max_decoder_steps,
-                temperature=temperature,
-                topk=topk,
-                use_cfg=use_cfg,
-                cfg_scale=cfg_scale,
-            )
+            if word_count >= longform_word_threshold:
+                # Longform path - process text - sentence by sentence
+                chunked_tokens, chunked_tokens_len, _ = chunk_and_tokenize_text_by_sentence(
+                    normalized_text, tokenizer_name, self.tokenizer, self.eos_id
+                )
+
+                chunk_state = self.create_longform_chunk_state(batch_size=1)
+                all_codes = []
+
+                for chunk_idx, (tokens, tokens_len) in enumerate(zip(chunked_tokens, chunked_tokens_len)):
+                    batch = {
+                        'text': tokens.unsqueeze(0).to(self.device),
+                        'text_lens': torch.tensor([tokens_len], device=self.device, dtype=torch.long),
+                        'speaker_indices': speaker_index,
+                    }
+                    end_of_text = [chunk_idx == len(chunked_tokens) - 1]
+                    beginning_of_text = chunk_idx == 0
+
+                    output = self.generate_long_form_speech(
+                        batch,
+                        chunk_state=chunk_state,
+                        end_of_text=end_of_text,
+                        beginning_of_text=beginning_of_text,
+                        max_decoder_steps=max_decoder_steps,
+                        temperature=temperature,
+                        topk=topk,
+                        use_cfg=use_cfg,
+                        apply_attention_prior=True,
+                        apply_prior_to_layers=apply_prior_to_layers,
+                        estimate_alignment_from_layers=estimate_alignment_from_layers,
+                        prior_epsilon=0.1,
+                        cfg_scale=cfg_scale,
+                    )
+                    if output.predicted_codes_lens[0] > 0:
+                        all_codes.append(output.predicted_codes[0, :, : output.predicted_codes_lens[0]])
+
+                # Concatenate and convert to audio
+                if all_codes:
```

Comment:
Can you make this if statement more detailed? Consider checking if the length is zero
In what circumstance is there no return from the generate_long_form_speech function?